api/src/main/java/eu/darkbot/api/game/entities/FakeEntity.java

```diff
@@ -0,0 +1,58 @@
+package eu.darkbot.api.game.entities;
+
+import eu.darkbot.api.game.other.Location;
+
+/**
+ * Fake entities help to keep virtual entities on a map that placed out of visibility range.
+ * It should use when need to assist or predict position these entities for active module.
+ * Cases of use: keeping mine locations for choose a better path and prevent releasing it,
+ *               keeping box locations that should be collected after more prefer tasks.
+ */
+public interface FakeEntity extends Entity {
+    static boolean isFakeEntity(Entity entity) {
+        return entity instanceof FakeEntity;
+    }
+
+    /**
+     * Set timeout lifetime of a fake object in ms, -1 - disable timer
+     */
+    void setTimeout(long keepAlive);
+
+    /**
+     * Set the minimum distance to a hero when an entity will be removed
+     */
+    void setRemoveDistance(long removeDistance);
+
+
+    /**
+     * Fake box is a target for colector module if more prefer boxes is absent
+     */
+    interface FakeBox extends Box, FakeEntity {
+        /**
+         * Is remove fake entity after attempt select it
+         */
+        void setRemoveWhenAttemptSelect(boolean removeWhenAttemptSelect);
+    }
+
+    /**
+     * Fake mine is an obstacle that counted by path builder
+     */
+    interface FakeMine extends Mine, FakeEntity {
+    }
+
+    /**
+     * Fake ship. It can be NPS, player or pet.
```

Comment:
NPC is a specific sub-entity, so you need to implement the Npc interface for that (ie: just add `FakeNpc extends FakeShip, Npc {`

Reply:
Fake ship required only for provide control on entity to user code. In main repo will be created `public class FakeNpc implement FakeShip, Npc {` that will be added to NPC list. By same way it will be implemented for `public class FakePlayer implement FakeShip, Player {` etc. But user code will be used FakeShip interface for managing it.